### DISCUSSION TITLE
Cardinal audio inputs 1-2 in Reaper act weird

### DISCUSSION THREAD

**Discussion by Abdusalyamov** [DISTRHO/Cardinal#719] [Q&A]
Hi!

I want to use cardinal as a vocoder with both carrier and modulator signals coming from other tracks, so I need 4 inputs. I insert Cardinal vst3 (not fx of synth because they don't have enough inputs) onto a track. The problem is even when inputs 1-2 are not connected to outputs 1-2 inside Cardinal, Reaper (v7.22) still plays the sound coming to inputs 1-2. Which as far as I understand it shouldn't. When I connect inputs 1-2 with outputs 1-2 inside cardinal the sound is adding up with itself. I tried messing with reaper routing but with no results.  Also tried CLAP version of Cardinal, it acts the same. So I'm asking for help here. How to make reaper not play sounds from inputs 1-2 right away if they are not connected to the outputs inside the plugin? (I've managed to find workaround, I send carrier signal into inputs to 5-6 instead of 1-2 but it feels kinda wrong, why shouldn't I be able to use inputs 1-2?)

**Comment by dromer** [comment]
Maybe you need to force multichannel? -> 
![2024-10-21_09-53](https://github.com/user-attachments/assets/362eb2e3-cd9a-437e-8d3c-0aa357254502)

I'm very new to Reaper and it's routing matrix, so it might be a limitation to how it works with multi-channel audio

You are using the 18 channel VSTi registered version?
![2024-10-21_09-54](https://github.com/user-attachments/assets/b030a512-6ee2-4d32-a6f0-09fcfe72be91)

Btw you do _not_ want to connect the CV Outputs to your speakers, so make sure they are muted or disconnected.
(Reaper does not properly support CV port features yet)

I see there is an issue with the LV2 that it registers as an FX and doesn't properly setup multi-channel. Seems we need to write some reports.

CLAP version only has stereo Instrument and FX versions, no multichannel i/o (it doesn't have CV ports yet)

**Reply by Abdusalyamov** [reply]
Thank you for the response, I tried forcing multichannel but it didn't do anything.
I'll try to add information to make it clearer:
As you can see on the image:
1. There's an audio coming through the first track
2. In audio host module nothing is connected 
3. Audio is sent to Cardinal's inputs 1 and 2


![image](https://github.com/user-attachments/assets/158af71c-21fd-4e2a-accb-3816497e7480)

**Comment by subderground** [comment]
Hi, I think it's the same ''problem'' I reported in the issues section : '''ins/outs in Cardinal 18ch'' #651 

>>As previous VST3 version, the dry audio goes thru Cardinal even if the cables are not connected.
I understand this thing for "Cardinal Instrument" version ; but the Cardinal "Standard" Main Version should be as near as possible with hardware modular stuffs.
The LV2 version works normally.
Thanks :-)<<

So the solution (or workaround) is to use the LV2 version.👍

Enjoy.
T.

**Reply by Abdusalyamov** [reply]
Thank you for the reply! I guess I'll have to use workaround, but I hope that developers will take a look at this issue and fix it
